internal/build/buildconfig/maker.go

```diff
@@ -0,0 +1,124 @@
+package buildconfig
+
+import (
+	"fmt"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	kmmv1beta1 "github.com/rh-ecosystem-edge/kernel-module-management/api/v1beta1"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/build"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+//go:generate mockgen -source=maker.go -package=buildconfig -destination=mock_maker.go
+
+type Maker interface {
+	MakeBuildConfig(mod kmmv1beta1.Module, buildConfig *kmmv1beta1.Build, targetKernel, containerImage string) (*buildv1.BuildConfig, error)
+}
+
+type maker struct {
+	helper build.Helper
+	scheme *runtime.Scheme
+}
+
+func NewMaker(helper build.Helper, scheme *runtime.Scheme) Maker {
+	return &maker{
+		helper: helper,
+		scheme: scheme,
+	}
+}
+
+func (m *maker) MakeBuildConfig(mod kmmv1beta1.Module, buildConfig *kmmv1beta1.Build, targetKernel, containerImage string) (*buildv1.BuildConfig, error) {
+	buildArgs := m.helper.ApplyBuildArgOverrides(
+		buildConfig.BuildArgs,
+		kmmv1beta1.BuildArg{Name: "KERNEL_VERSION", Value: targetKernel},
+	)
+
+	bc := buildv1.BuildConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: mod.Name + "-",
+			Namespace:    mod.Namespace,
+			Labels:       build.GetBuildLabels(mod, targetKernel),
+		},
+		Spec: buildv1.BuildConfigSpec{
+			Triggers: []buildv1.BuildTriggerPolicy{
+				{Type: buildv1.ConfigChangeBuildTriggerType},
+			},
+			RunPolicy: buildv1.BuildRunPolicySerialLatestOnly,
+			CommonSpec: buildv1.CommonSpec{
+				ServiceAccount: builderServiceAccountName,
+				Source: buildv1.BuildSource{
+					Dockerfile: &buildConfig.Dockerfile,
+					Type:       buildv1.BuildSourceDockerfile,
+				},
+				Strategy: buildv1.BuildStrategy{
+					Type: buildv1.DockerBuildStrategyType,
+					DockerStrategy: &buildv1.DockerBuildStrategy{
+						BuildArgs: EnvVarsFromKMMBuildArgs(buildArgs),
+						Volumes:   BuildVolumesFromBuildSecrets(buildConfig.Secrets),
+					},
+				},
+				Output: buildv1.BuildOutput{
```

Comment:
Are we going to use ImageStream? Is it next stage?

Reply:
The user only specifies a container image name as destination; it is not easy for us to determine if that image name is from an ImageStream or not.
The usual authentication against the ImageStream works even though you are using an image name, so I don't think there is any downside right now.

Reply:
So customer will be responsible for creating image stream, right?

Reply:
Yes.